lib/src/builder/build_step_impl.dart

```diff
@@ -87,4 +94,11 @@ class BuildStepImpl implements BuildStep {
   /// Resolves [id] and returns a [Future<Resolver>] once that is done.
   Future<Resolver> resolve(AssetId id) async => new Resolver(
       await _resolvers.get(toBarbackTransform(this), [toBarbackAssetId(id)]));
+
+  /// Should be called after [build] has completed. This will wait until for
+  /// [_outputsCompleted] and will also close the [logger].
+  Future finalize() async {
```

Comment:
This name may be confusing for Java coders. Maybe `finish`, `complete`, `releaseResources` or similar